File: union.js
class UNION {
    constructor(a = [], b = []) {
        this.a = a;
        this.b = b;
        this.result = [];

        this.aSize = a.length;
        this.bSize = b.length;
        this.i = 0;
        this.j = 0;
        this.node1 = "";

        this.con = true;
    }

    update() {
        let aLast = this.aSize - 1;
        let bLast = this.bSize - 1;

        if (((this.i < this.aSize) || (this.j < this.bSize)) && this.con) {
            switch (true) {
                case aLast < this.i:
                    this.node1 = "Az \"a\" halmaznak nincs több vizsgálható eleme, ezért belerakjuk a \"b\" összes többi elemét az únióba";
                    for (let k = this.j; k < this.bSize; k++) {
                        this.result.push(this.b[k]);
                    }
                    this.con = false;
                    break;

                case bLast < this.j:
                    this.node1 = "Az \"b\" halmaznak nincs több vizsgálható eleme, ezért belerakjuk a \"a\" összes többi elemét az únióba";
                    for (let k = this.i; k < this.aSize; k++) {
                        this.result.push(this.a[k]);
                    }
                    this.con = false;
                    break;

                case this.a[this.i] == this.b[this.j]:
                    this.node1 = "Mindkét elem azonos, ezért egyik sem kissebb, mint a másik. Mindkét pointert továbbléptetjük";
                    this.result.push(this.a[this.i]);
                    this.j++;
                    this.i++;
                    break;

                case this.a[this.i] < this.b[this.j]:
                    this.node1 = "\"a\" jelen eleme kissebb, mint \"b\" eleme, ezért a jelenlegi elemét hozzáadjuk és továbbléptetjük a Pointert"
                    this.result.push(this.a[this.i]);
                    this.i++;
                    break;

                case this.b[this.j] < this.a[this.i]:
                    this.node1 = "\"b\" jelen eleme kissebb, mint \"a\" eleme, ezért a jelenlegi elemét hozzáadjuk és továbbléptetjük a Pointert"
                    this.result.push(this.b[this.j]);
                    this.j++;
                    break;
            }
        }
    }

    show() {
        return {
            pointerA: this.i,
            pointerB: this.j,
            comparison: this.node1,
            currentResult: this.result,
            setA: this.a,
            setB: this.b,
        }
    }
}

File: union.java
import java.util.ArrayList;

public class union {
    public static void main(String[] args) {
        System.out.println(uni_rendezett(
            new int[] {-1, 0, 2, 5, 9},
            new int[] { 0, 2, 4, 5, 6, 7, 9}
        ));

        System.out.println(uni_nemrendezett(
            new int[] {-1, 0, 2, 5, 9},
            new int[] { 0, 2, 4, 5, 6, 7, 9}
        ));
    }
    public static ArrayList < Integer > uni_rendezett(int[] a, int[] b) {
        ArrayList < Integer > result = new ArrayList < Integer > ();
        int aSize = a.length;
        int bSize = b.length;

        for (int i = 0, j = 0; (i < aSize) || (j < bSize);) {
            if (aSize-1 < i) {
                for (int k = j; k < bSize; k++) {
                    result.add(b[k]);
                }
                break;
            }

            if (bSize-1 < j) {
                for (int k = i; k < aSize; k++) {
                    result.add(a[k]);
                }
                break;
            }

            if (a[i] == b[j]) {
                result.add(a[i]);
                j++;
                i++;
                continue;
            }

            if (a[i] < b[j]) {
                result.add(a[i]);
                i++;
                continue;
            }

            if (b[j] < a[i]) {
                result.add(b[j]);
                j++;
                continue;
            }
        }
        return result;
    }

    public static ArrayList < Integer > uni_nemrendezett(int[] a, int[] b) {
        ArrayList < Integer > result = new ArrayList < Integer > ();
        int[] biggerList = a.length > b.length ? a : b;
        int[] smallerList = a.length > b.length ? b : a;

        for (int i = 0; i < biggerList.length; i++) {
            result.add(biggerList[i]);
        }

        for (int i = 0; i < smallerList.length; i++) {
            if (!(result.contains(smallerList[i]))) {
                result.add(smallerList[i]);
            }
        }


        return result;
    }
}
